src/ziggurat/middleware/batch/batch_proto_deserializer.clj

```diff
@@ -0,0 +1,19 @@
+(ns ziggurat.middleware.batch.batch-proto-deserializer
+  (:require [ziggurat.middleware.default :refer [deserialize-message]]))
+
+(defn- deserialize-key-and-value
+  [key-proto-class value-proto-class topic-entity]
+  (fn [message]
+    (let [key                (:key message)
+          value              (:value message)
+          deserialized-key   (when (some? key) (deserialize-message key key-proto-class (name topic-entity)))
+          deserialized-value (when (some? value) (deserialize-message value value-proto-class (name topic-entity)))]
```

Comment:
the (when ...) expressions will still return nil if `(some? key)` or `(some? value)` is false

Reply:
That's the intention, actually. Just return nil if a nil value is encountered and not attempt a deserialization.

WDYT @macalimlim ?

Reply:
if that happened, are we expecting the function to return a map with a nil `:key` and `:value`?